model_zoo/lbann.cpp

```diff
@@ -89,7 +89,8 @@ int main(int argc, char *argv[]) {
     lbann_data::Model *pb_model = pb.mutable_model();
 
     auto model = build_model_from_prototext(argc, argv, pb_trainer, pb,
-                                            comm.get(), opts, io_thread_pool, true);
+                                            comm.get(), opts, trainer->get_name(), io_thread_pool,
```

Comment:
IMO, this is the wrong way to associate a model with a trainer... Why the string when you actually have a trainer object?

Reply:
Also, the function name is `build_model_from_prototext`. There's so much garbage in this call that is *not* related to constructing `model` objects from prototext descriptions of them... This feels like adding more of that rather than clarifying the interface.